src/twisted/web/test/test_web.py

```diff
@@ -1849,3 +1852,50 @@ def test_defaultReactor(self):
 
         factory = http.HTTPFactory()
         self.assertIs(factory.reactor, reactor)
+
+
+class QueueResource(Resource):
+    """
+    Add all requests to an internal queue,
+    without responding to the requests.
+    You can access the requests from the queue and handle their response.
+    """
+
+    isLeaf = True
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.dispatchedRequests: List[Request] = []
+
+    def render_GET(self, request: Request) -> int:
+        self.dispatchedRequests.append(request)
+        return NOT_DONE_YET
+
+
+class TestRFC9112Section932(unittest.TestCase):
+    """
+    Verify that HTTP/1.1 request ordering is preserved.
+    """
+
+    def test_multipleRequestsInOneSegment(self) -> None:
+        """
+        Twisted MUST NOT respond to a second HTTP/1.1 request while the first
+        is still pending.
+        """
+        qr = QueueResource()
+        site = Site(qr)
+        proto = site.buildProtocol(None)
+        serverTransport = StringTransport()
+        proto.makeConnection(serverTransport)
+        proto.dataReceived(
+            b"GET /first HTTP/1.1\r\nHost: a\r\n\r\n"
+            b"GET /second HTTP/1.1\r\nHost: a\r\n\r\n"
+        )
+        # The TCP data contains 2 requests,
+        # but only 1 request was dispatched,
+        # as the first request was not yet finalized.
+        self.assertEqual(len(qr.dispatchedRequests), 1)
+        # The first request is finalized and the
+        # second request is dispatched right away.
+        qr.dispatchedRequests[0].finish()
+        self.assertEqual(len(qr.dispatchedRequests), 2)
```

Comment:
I think that is best to have the new test added to `t.w.test.test_http.PipeliningBodyTests`

But this tests is still kept in test_web,
maybe it is worth adding the test in which the request is received in a separate package.

I know that this test is green even on `trunk`, but maybe it can help with future refactoring.

```suggestion
        self.assertEqual(len(qr.dispatchedRequests), 2)

    def test_multipleRequestsInDifferentSegments(self) -> None:
        """
        Twisted MUST NOT respond to a second HTTP/1.1 request while the first
        is still pending, even if the second request is received in a separate
        TCP package.
        """
        qr = QueueResource()
        site = Site(qr)
        proto = site.buildProtocol(None)
        serverTransport = StringTransport()
        proto.makeConnection(serverTransport)
        raw_data = (
            b"GET /first HTTP/1.1\r\nHost: a\r\n\r\n"
            b"GET /second HTTP/1.1\r\nHost: a\r\n\r\n"
        )
        # Just go byte by byte for the extreme case in which each byte is
        # received in a separate TCP package.
        for chunk in iterbytes(raw_data):
            proto.dataReceived(chunk)
        # The TCP data contains 2 requests,
        # but only 1 request was dispatched,
        # as the first request was not yet finalized.
        self.assertEqual(len(qr.dispatchedRequests), 1)
        # The first request is finalized and the
        # second request is dispatched right away.
        qr.dispatchedRequests[0].finish()
        self.assertEqual(len(qr.dispatchedRequests), 2)

```